File: app/src/main/java/com/productions/gizzmoo/pokemonpuzzleleague/settings/SavedState.java
package com.productions.gizzmoo.pokemonpuzzleleague.settings;

import android.os.Parcel;
import android.os.Parcelable;
import android.preference.Preference;

/**
 * Created by Chrystian on 4/11/2018.
 */

public class SavedState extends Preference.BaseSavedState {
    // Member that holds the setting's value
    // Change this data type to match the type saved by your Preference
    int value;

    public SavedState(Parcelable superState) {
        super(superState);
    }

    public SavedState(Parcel source) {
        super(source);
        value = source.readInt();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
        dest.writeInt(value);
    }

    // Standard creator object using an instance of this class
    public static final Parcelable.Creator<SavedState> CREATOR =
            new Parcelable.Creator<SavedState>() {

                public SavedState createFromParcel(Parcel in) {
                    return new SavedState(in);
                }

                public SavedState[] newArray(int size) {
                    return new SavedState[size];
                }
            };
}

File: app/src/main/java/com/productions/gizzmoo/pokemonpuzzleleague/settings/SavedState.kt
package com.productions.gizzmoo.pokemonpuzzleleague.settings

import android.os.Parcel
import android.os.Parcelable
import android.preference.Preference

class SavedState(superState: Parcelable) : Preference.BaseSavedState(superState) {
    internal var value: Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
        dest.writeInt(value)
    }
}
